samples/common/src/consumer_config.rs

```diff
@@ -12,8 +12,8 @@ const CONFIG_FILENAME: &str = "consumer_settings";
 #[derive(Debug, Deserialize)]
 pub struct Settings {
     pub consumer_authority: Option<String>,
-    pub chariott_url: Option<String>,
-    pub invehicle_digital_twin_url: Option<String>,
+    pub chariott_uri: Option<String>,
```

Comment:
I think you should enforce the user to provide either the In-vehicle URI or Chariott URI, but not both at the same time.

You can try doing what William had suggested in a past PR for Freyja (https://github.com/eclipse-ibeji/freyja/pull/20/files/4a27591f6f0bb139dafbe3162c4192136fa49e91#r1268597172). This struct would be an enum in that case.

Reply:
This comes down to personal preferences.  Leaving it as is.